presto-main/src/test/java/com/facebook/presto/sql/query/QueryAssertions.java

```diff
@@ -66,6 +66,8 @@ public void assertQuery(@Language("SQL") String actual, @Language("SQL") String
             fail("Execution of 'expected' query failed: " + expected, ex);
         }
 
+        assertEquals(actualResults.getTypes(), expectedResults.getTypes(), "For query: \n " + actual + "\n:");
```

Comment:
I would change the message a bit to: `"Types mismatch for query: \n " + actual + "\n:"`